### PR TITLE
Update onnx proto and onnx parser to handle fp8 types

### PR DESCRIPTION
**Pull request by umangyadav**
updates `onnx.proto` files to be same as `onnx==1.14.1` that is used by MIGraphX. File is copied from [here](https://github.com/onnx/onnx/blob/rel-1.14.1/onnx/onnx.proto).
Add more switch cases in onnx parser to handle FP8 dtypes. 

Testing : Reading of quantized fp8 onnx model works. it runs on "ref" alright as well.

Question : Are there any backend tests that need to be enabled ?

### CODE REVIEW

File: src/onnx/onnx_parser.cpp [hunk 1]
```diff
@@ -609,6 +625,13 @@ shape::type_t get_type(int dtype)
     case 11: return shape::double_type;
     case 12: return shape::uint32_type;
     case 13: return shape::uint64_type;
+    case 18: return shape::fp8e4m3fnuz_type;
+    case 14:
+    case 15:
+    case 16:
+    case 17:
+    case 19:
+    case 20:
```
Comment (CharlieL7) [inline src/onnx/onnx_parser.cpp:634]: Why have these if we don't support those datatypes? We also don't support sparse_tensor_type which is 8.

Comment (umangyadav) [reply src/onnx/onnx_parser.cpp:634]: I have only added `E4M3FNUZ` type inside the MIGraphX IR so far. 

That's why only `case 18` is supported.

Comment (umangyadav) [reply src/onnx/onnx_parser.cpp:634]: `sparse_tensor_type` is for something else. It is TypeProto something not related to data type.

**Comment by attila-dusnoki-htec** [comment]
> Question : Are there any backend tests that need to be enabled ?

There are some:
https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/test/py/onnx_backend_test.py#L871-L921

**Comment by umangyadav** [comment]
> > Question : Are there any backend tests that need to be enabled ?
> 
> There are some: https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/test/py/onnx_backend_test.py#L871-L921

The other things is that  many of the tests are cast and castlike. Which would require some update in parsing logic of MIGraphX to account for new "saturate" attribute.

**Comment by pfultz2** [comment]
#2504 should fix the format issue.

**Review by CharlieL7 — APPROVED** [review]

**Pull request merged**